sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2ScanExec.scala

```diff
@@ -93,21 +81,17 @@ case class DataSourceV2ScanExec(
         sparkContext,
         sqlContext.conf.continuousStreamingExecutorQueueSize,
         sqlContext.conf.continuousStreamingExecutorPollIntervalMs,
-        partitions).asInstanceOf[RDD[InternalRow]]
-
-    case r: SupportsScanColumnarBatch if r.enableBatchRead() =>
-      new DataSourceRDD(sparkContext, batchPartitions).asInstanceOf[RDD[InternalRow]]
+        partitions,
+        schema,
+        partitionReaderFactory.asInstanceOf[ContinuousPartitionReaderFactory])
```

Comment:
This should not cast. Just call `readSupport.createContinuousReaderFactory(...)` here.

Reply:
`DataSourceV2ScanExec` is shared between batch and streaming, so the `partitionReaderFactory` here is a general type instead of the concrete `ContinuousPartitionReaderFactory`. I think we can avoid this cast in the future refactoring, when we have a dedicated scan plan for continuous streaming.

Reply:
However you want to do it is fine with me, but I've seen excessive casting in the SQL back-end so I'm against adding it when it isn't necessary, like this case.